ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/KnownPeers.hs

```diff
@@ -77,6 +78,12 @@ belowTarget actions
                              canAsk
                              numPeerShareReqsPossible
       let numPeerShareReqs = Set.size selectedForPeerShare
+          numPeersToReq    = (targetNumberOfKnownPeers - numKnownPeers)
+                             `div`
+                             numPeerShareReqs
+          selForPSWithAmount = Set.toList
+                             $ Set.map (\a -> (fromIntegral numPeersToReq, a))
+                                       selectedForPeerShare
```

Comment:
comment that we are splitting the amount of peers to request across the selected peers for diversity